bootloader/firmware/CMakeLists.txt

```diff
@@ -0,0 +1,125 @@
+# This CMakeLists.txt handles everything that is compiled only when
+# cross-compiling, like the board support packages and special options.
+find_package(STM32G4xx)
+add_STM32G4_driver("Bootloader")
+
+set(COMMON_EXECUTABLE_DIR "${CMAKE_SOURCE_DIR}/common/firmware")
+
+# Add source files that should be checked by clang-tidy here
+set(BOOTLOADER_FW_LINTABLE_SRCS
+        ${CMAKE_CURRENT_SOURCE_DIR}/main.c
+        )
+
+set(BOOTLOADER_FW_NON_LINTABLE_SRCS
+        ${CMAKE_CURRENT_SOURCE_DIR}/system_stm32g4xx.c
+        ${CMAKE_CURRENT_SOURCE_DIR}/stm32g4xx_it.c
+        ${CMAKE_CURRENT_SOURCE_DIR}/clocking.c
+        ${CMAKE_CURRENT_SOURCE_DIR}/can.c
+        ${COMMON_EXECUTABLE_DIR}/errors/errors.c)
+
+add_executable(bootloader
+        ${BOOTLOADER_FW_LINTABLE_SRCS}
+        ${BOOTLOADER_FW_NON_LINTABLE_SRCS})
+
+target_link_libraries(bootloader
+        PUBLIC STM32G491RETx
```

Comment:
I believe that this is incorrect. `STM32G491RETx` is the package for our application subsystems. 

The bootloader needs its own set of files, right?

Reply:
Yeah. It will need a different linker script and maybe a different startup assembly